pkg/providers/aws/provider_redis.go

```diff
@@ -158,6 +159,66 @@ func (p *AWSRedisProvider) createElasticacheCluster(ctx context.Context, r *v1al
 		return nil, "modify elasticache cluster in progress", nil
 	}
 
+	// Add Tags to AWS Elasticache
+	if err != nil {
+		logrus.Errorf("Error Failed to get strategy config: %s", err)
+	}
+	cacheSvcTag := elasticache.New(session.Must(session.NewSession(&aws.Config{Region: aws.String(stratCfg.Region)})))
+	// need to loop through each redis instance
+	cacheInstance := *foundCache.NodeGroups[0]
+	for _, cache := range cacheInstance.NodeGroupMembers {
+
+		cacheRegion := *cache.PreferredAvailabilityZone
+		regionlen := len(cacheRegion)
+		cacheRegion = cacheRegion[:regionlen-1]
+		cacheClusterId := *cache.CacheClusterId
+		// get the account number
+		svc := sts.New(session.New())
+		inputSts := &sts.GetCallerIdentityInput{}
+		id, err := svc.GetCallerIdentity(inputSts)
+		accountId := *id.Account
+		// need arn in the following format arn:aws:elasticache:us-east-1:1234567890:cluster:my-mem-cluster
+		arn := "arn:aws:elasticache:" + cacheRegion + ":" + accountId + ":cluster:" + cacheClusterId
+
+		// Set the Tag values
+		defaultOrganizationTag := r.Spec.Env.Value
+		clusterId, _ := resources.GetClusterId(ctx, p.Client)
+		cacheTag := []*elasticache.Tag{
+			{
+				Key:   aws.String(defaultOrganizationTag + "clusterId"),
+				Value: aws.String(clusterId),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "resource-type"),
+				Value: aws.String(r.Spec.Type),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "resource-name"),
+				Value: aws.String(r.Name),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "product-name"),
+				Value: aws.String(r.Spec.Labels.ProductName),
+			},
+		}
+		input := &elasticache.AddTagsToResourceInput{
+			ResourceName: aws.String(arn),
+			Tags:         cacheTag,
+		}
+		// add tags
+		_, err = cacheSvcTag.AddTagsToResource(input)
+
+		//TODO when snapshots are created add tags to them
+		//inputDescirbe := &elasticache.DescribeSnapshotsInput{
```

Comment:
Do we need to leave this commented out code?

Reply:
I just noticed this work is done here https://github.com/integr8ly/cloud-resource-operator/pull/75

Can we look at merging these together, to speed up the verification

Reply:
As they are separate jira's, I believe they should remain separate. Not in favour of large pr's, they tend to make tracking changes difficult and also reverting.